tests/beman/inplace_vector/spec.test.cpp

```diff
@@ -0,0 +1,231 @@
+// 1 An inplace_vector is a contiguous container. Its capacity is fixed and its
+// elements are stored within the inplace_vector object itself.
+//
+// This is not testable.
+
+// 2 An inplace_vector meets all of the requirements of a container (23.2.2.2),
+// of a reversible container (23.2.2.3), of a contiguous container, and of a
+// sequence container, including most of the optional sequence container
+// requirements (23.2.4). The exceptions are the push_front, prepend_range,
+// pop_front, and emplace_front member functions, which are not provided.
+// Descriptions are provided here only for operations on inplace_- vector that
+// are not described in one of these tables or for operations where there is
+// additional semantic information.
+//
+// TODO: Test container & reversible container & optional sequence container
+// req.
+
+// 3 For any N, inplace_vector<T, N>::iterator and inplace_vector<T,
+// N>::const_iterator meet the constexpr iterator requirements.
+//
+// TODO: Test this.
+
+// 4 For any N > 0, if is_trivial_v<T> is false, then no inplace_vector<T, N>
+// member functions are usable in constant expressions.
+//
+// This would be tested in subsequent method tests.
+
+// 5 Any member function of inplace_vector<T, N> that would cause the size to
+// exceed N throws an exception of type bad_alloc.
+//
+// TODO: Test this.
+
+// 6 Let IV denote a specialization of inplace_vector<T, N>. If N is zero, then
+// IV is both trivial and empty. Otherwise:
+// (6.1) — If is_trivially_copy_constructible_v<T> is true, then IV has a
+// trivial copy constructor
+// (6.2) — If is_trivially_move_constructible_v<T> is true, then IV has a
+// trivial move constructor.
+// (6.3) — If is_trivially_destructible_v<T> is true, then:
+// (6.3.1) — IV has a trivial destructor.
+// (6.3.2) — If is_trivially_copy_constructible_v<T> &&
+// is_trivially_copy_assignable_v<T> is true, then IV has a trivial copy
+// assignment operator.
+// (6.3.3) — If is_trivially_move_constructible_v<T> &&
+// is_trivially_move_assignable_v<T> is true, then IV has a trivial move
+// assignment operator.
+
+TEST(Overview, Section6) {
+  // Let IV denote a specialization of inplace_vector<T, N>.
+  {
+    SCOPED_TRACE("If N is zero, then IV is both trivial and empty.");
+
+    struct T {
+      T() = delete;
+      ~T() {}
+
+      T(T &) = delete;
+      T &operator=(T &) = delete;
+
+      T(T &&) = delete;
+      T &operator=(T &&) = delete;
+    };
+    static_assert(!std::is_trivial_v<T>);
+    using IV = inplace_vector<T, 0>;
+    EXPECT_TRUE(std::is_trivial_v<IV> && std::is_empty_v<IV>);
+  }
+  {
+    SCOPED_TRACE(
+        "(6.1) — If is_trivially_copy_constructible_v<T> is true, then "
+        "IV has a trivial copy constructor");
+
+    struct T {
+      int a = 0;
+      int b;
+
+      T(T const &) = default;
+
+      T() = delete;
+      T(T &&) = delete;
+      // TODO: This type is also trivially destructible
+      // ~T() {}
```

Comment:
Question: Is it possible to construct a trivially copy constructable struct that is not trivially destructable ?

Reply:
Answer: no,
> is_trivially_copy_constructible is the same as is_trivially_constructible<T, const T&>, which in turn is requires is_constructible<T, const T&> where all operations are known to be trivial. is_constructible is defined in https://eel.is/c++draft/type.traits#meta.unary.prop-9, which refers to a variable, so the destructor is included.

Reply:
```suggestion
      // trivially copy constructible implies trivially destructible
```